### PR TITLE
expression:fix FORMAT compatibility issue #11206

### LINKED ISSUE #11206: Results of function FORMAT is different from MySQL.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```mysql
SELECT FORMAT(12332.2, 2, 'de_DE'); show warnings;
SELECT FORMAT(12332.2, 2, 'xxxxx'); show warnings;
```

2. What did you expect to see?

```mysql
mysql> SELECT FORMAT(12332.2, 2, 'de_DE'); show warnings;
+-----------------------------+
| FORMAT(12332.2, 2, 'de_DE') |
+-----------------------------+
| 12.332,20                   |
+-----------------------------+
1 row in set (0.00 sec)

Empty set (0.00 sec)

mysql> SELECT FORMAT(12332.2, 2, 'xxxxx'); show warnings;
+-----------------------------+
| FORMAT(12332.2, 2, 'xxxxx') |
+-----------------------------+
| 12,332.20                   |
+-----------------------------+
1 row in set, 1 warning (0.00 sec)

+---------+------+-------------------------+
| Level   | Code | Message                 |
+---------+------+-------------------------+
| Warning | 1649 | Unknown locale: 'xxxxx' |
+---------+------+-------------------------+
1 row in set (0.00 sec)
```

3. What did you see instead?

```mysql
mysql> SELECT FORMAT(12332.2, 2, 'de_DE'); show warnings;
ERROR 1105 (HY000): not support for the specific locale
Empty set (0.00 sec)

mysql> SELECT FORMAT(12332.2, 2, 'xxxxx'); show warnings;
ERROR 1105 (HY000): not support for the specific locale
Empty set (0.00 sec)
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
8c20289c7

## SIG slack channel
[#sig-exec](https://tidbcommunity.slack.com/archives/CMRD79DRR)

## Score
- 300

## Mentor
 * @XuHuaiyu

### PR DESCRIPTION
**Pull request by xiaofeitu2**
### What problem does this PR solve?

Issue Number: close #11206

Problem Summary:
some unsurported locale may cause error and it is different from mysql.default en_US and  warning is expected 

### What is changed and how it works?

in builtin_string.go function evalString add an if logic to make it en_US default and not display the error

### Check List
Tests

- Unit test
modify the test func TestFormat

### Release note

- No release note

### CODE REVIEW

**Review by zz-jason — COMMENTED** [review]
@xiaofeitu2 please add some unit tests.

File: expression/builtin_string.go [hunk 1]
```diff
@@ -3368,10 +3368,15 @@ func (b *builtinFormatWithLocaleSig) evalString(row chunk.Row) (string, bool, er
 	if err != nil {
 		return "", false, err
 	}
+	isSurportLocale := locale == "en_US" || locale == "zh_CN"
```
Comment (rebelice) [inline expression/builtin_string.go:3371]: It's better to use map `locale2FormatFunction` to check if it support.

File: expression/builtin_string_test.go [hunk 2]
```diff
@@ -1825,8 +1825,11 @@ func (s *testEvaluatorSuite) TestFormat(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(f3, NotNil)
 	r3, err := evalBuiltinFunc(f3, chunk.Row{})
-	c.Assert(types.NewDatum(err), testutil.DatumEquals, types.NewDatum(errors.New("not support for the specific locale")))
+	c.Assert(err, IsNil)
 	c.Assert(r3, testutil.DatumEquals, types.NewDatum(formatTests3.ret))
+	showwarnings := s.ctx.GetSessionVars().StmtCtx.GetWarnings()
```
Comment (rebelice) [inline expression/builtin_string_test.go:1830]: Could you please use `showWarnings` instead of `showwarnings`?

**Comment by wjhuang2016** [comment]
@xiaofeitu2 Ping, could you update your PR?

**Comment by zz-jason** [comment]
I'm going to close this PR since the original issue has been fixed, thanks for your contribution!

**Pull request closed without merging**